src/check/property/AsyncProperty.generic.ts

```diff
@@ -3,14 +3,17 @@ import { Arbitrary } from '../arbitrary/definition/Arbitrary';
 import { Shrinkable } from '../arbitrary/definition/Shrinkable';
 import { PreconditionFailure } from '../precondition/PreconditionFailure';
 import { IRawProperty, runIdToFrequency } from './IRawProperty';
+import { readConfigureGlobal, GlobalAsyncPropertyHookFunction } from '../runner/configuration/GlobalParameters';
 
 /**
  * Type of legal hook function that can be used to call `beforeEach` or `afterEach`
  * on a {@link IAsyncPropertyWithHooks}
  *
  * @public
  */
-export type AsyncPropertyHookFunction = (() => Promise<unknown>) | (() => void);
+export type AsyncPropertyHookFunction =
+  | ((globalHookFunction: GlobalAsyncPropertyHookFunction) => Promise<unknown>)
+  | ((globalHookFunction: GlobalAsyncPropertyHookFunction) => void);
```

Comment:
```suggestion
export type AsyncPropertyHookFunction =
  | ((previousHookFunction: AsyncPropertyHookFunction) => Promise<unknown>)
  | ((previousHookFunction: AsyncPropertyHookFunction) => void);
```

Reply:
`previousHookFunction` can't be an `AsyncPropertyHookFunction`. This function doesn't accept any input arguments. That's why I used `GlobalAsyncPropertyHookFunction` instead.